openedx/core/djangoapps/content/search/documents.py

```diff
@@ -485,6 +486,19 @@ def searchable_doc_for_collection(
     if collection:
         assert collection.key == collection_key
 
+        draf_num_children = 0
+        published_num_children = 0
+
+        # Count draft and published children
+        for entity in collection.entities.all():
+            draft = authoring_api.get_draft_version(entity.id)
+            published = authoring_api.get_published_version(entity.id)
+
+            if draft:
+                draf_num_children += 1
+            if published:
+                published_num_children += 1
```

Comment:
@ChrisChV CC @ormsbee

If a collection contains 200 items, will this result in 400 queries? I think we need a way to just get this as

```suggestion
        draft_num_children = collection.entities.filter(has_draft).count()
        published_num_children =collection.entities.filter(has_published).count()
```

Is that `filter` possible?

Reply:
That filter is definitely possible, but we might want to split it out a bit to give the publishing API control over translating from PublishableEntities -> Drafts... Something like

```python
draft_num_children = get_drafts(collection.entities.all()).count()
```

Where `get_drafts` is implemented in the `publishing` app. So ostensibly anything can throw in its qset of PublishableEntities and get Drafts without `collections` needing to know how that mapping really works.

Reply:
Updated https://github.com/openedx/edx-platform/pull/35734/commits/b6f2d98a02bdf5e2c3be7062685b0f56a3b2bfee, using https://github.com/openedx/openedx-learning/pull/257